src/main/flight/pid.c

```diff
@@ -248,7 +248,7 @@ void pidController(const pidProfile_t *pidProfile, const rollAndPitchTrims_t *an
         // -----calculate P component and add Dynamic Part based on stick input
         float PTerm = Kp[axis] * errorRate * tpaFactor;
         if (axis == FD_YAW) {
-            PTerm = ptermYawFilterApplyFn(ptermYawFilter, PTerm);
+            PTerm = yawNotchFilterApplyFn(yawFilterNotch, PTerm);
```

Comment:
@borisbstyle , I guess I didn't explain myself very well. So instead of doing more explaining, I've done a PR with how I think the filter setup should be. See PR https://github.com/betaflight/betaflight/pull/2336

Note that the only place `pid.c` needed to change was in `pidInitFilters` - this was specifically designed so that the filters could be changed without impacting the rest of the code.

As a bonus, I've changed it so that you can set either a PT1 or a notch filter on yaw. If the cutoff frequency is specified as zero, then a PT1 filter is used. If both frequencies are zero, then no filter is used.